Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -1819,3 +1824,15 @@ extension BrowserViewController: ContextMenuHelperDelegate {
             }
     }
 }
+
+extension BrowserViewController: HashchangeHelperDelegate {
+    func hashchangeHelperDidHashchange(hashchangeHelper: HashchangeHelper) {
+        if let tab = tabManager.selectedTab,
+           let webView = tab.webView {
+                urlBar.currentURL = tab.displayURL
+                toolbar?.updateBackStatus(webView.canGoBack)
+                toolbar?.updateFowardStatus(webView.canGoForward)
+            }
```

Comment:
Nit: unindent this body by one level (should only be tabbed once after the `if`). The alignment does look a bit close the `let`s, so maybe it would help to put the `{` on a new line here. Example:

```
if let tab = tabManager.selectedTab,
   let webView = tab.webView
{
    urlBar.currentURL = tab.displayURL
    ...
}
```